README.md

```diff
@@ -1,14 +1,397 @@
-the rights to use your contribution. For details, visit https://cla.opensource.microsoft.com.
+This library contains a set of reusable classes useful in ASP.NET Core:
 
-When you submit a pull request, a CLA bot will automatically determine whether you need to provide
-a CLA and decorate the PR appropriately (e.g., status check, comment). Simply follow the instructions
-provided by the bot. You will only need to do this once across all repos using our CLA.
+- [Web applications](#web-apps) that sign in users, and, optionally, call web APIs
+- [Protected web APIs](#web-apis), which, optionally call downstream web APIs
 
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
-For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
-contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+to enable them to work with the Microsoft identity platform (formerly named Azure AD v2.0 endpoint). In the library, web apps and protected web APIs are collectively referred to as web resources.
+
+
+ ## Breaking changes
+
+ If you've been using Microsoft.Identity.Web in your project you might want to know that, on 12/05/2019, a number of APIs are renamed to be consistent:
+.AddMicrosoftIdentityPlatformAuthentication => AddSignIn
+.AddMsal => .AddWebAppCallsProtectedWebApi
+.AddProtectedWebApiCallsWebAPis => AddProtectedWebApiCallsProtectedWebAPi
+
+## Web apps
+
+As of today, ASP.NET Core web apps templates (`dot net new mvc -auth`) create web apps that sign in users with the Azure AD v1.0 endpoint (allowing to sign in users with their organizational accounts, also named *Work or school accounts*). This library brings `ServiceCollection` extension methods to be used in the ASP.NET Core web app **Startup.cs** file to enable the web app to sign in users with the Microsoft identity platform (formerly Azure AD v2.0 endpoint), and, optionally enable the web app to call APIs on behalf of the signed-in user.
+
+![WebAppServiceCollectionExtensions](https://user-images.githubusercontent.com/13203188/64252959-82ae3680-cf1c-11e9-8a01-0a0be728a78e.png)
+
+### Web apps that sign in users - Startup.cs
+
+To enable users to sign in with the Microsoft identity platform, you need to replace, in your web application Startup.cs file, the following code:
+
+```CSharp
+using Microsoft.Identity.Web;
+
+public class Startup
+{
+  ...
+  public void ConfigureServices(IServiceCollection services)
+  {
+   ...
+   services.AddAuthentication(AzureADDefaults.AuthenticationScheme)
+           .AddAzureAD(options => Configuration.Bind("AzureAd", options));
+   ...
+  }
+  ...
+}
+```
+
+by
+
+```CSharp
+using Microsoft.Identity.Web;
+
+public class Startup
+{
+  ...
+  public void ConfigureServices(IServiceCollection services)
+  {
+   ...
+   services.AddSignIn(Configuration);

```

Comment:
Chris Ross recommended we also add the `services.AddAuthentication(AzureADDefaults.AuthenticationScheme)` before. Something to think of